pkg/client/ghcr/path.go

```diff
@@ -1,20 +1,21 @@
 package ghcr
 
 import (
-	"regexp"
 	"strings"
 )
 
-var (
-	reg = regexp.MustCompile(`^ghcr.io$`)
-)
-
 func (c *Client) IsHost(host string) bool {
-	return reg.MatchString(host)
+	return host == "ghcr.io"
 }
```

Comment:
I get that this makes it simpler, but does make it harder or more complex if github decide to add additional subdomains etc or even their domain (not sure they would) additionally this does some what diverge the logic from other clients.

Reply:
I don't think the fact that we might need a regex down the line is a good reason to use a regex unnecessarily here. If GHCR starts using different domains, we could introduce a regex when that happens. We'd have to update the code either way in that situation.

I also don't think the implementation matching how things are done in other clients has much benefit either. As long as the interface is satisfied, I think the details of the implementation should be the simplest possible.

That being said, I just made this change in passing and it isn't really related to the PR, so happy to revert. It's more of a style thing than anything else.

Reply:
To a Go newby like me the newer version is easier to understand in the context of the file it is in.
I agree, if GitHub adds new domains, we'll likely want an issue and code change down the line anyway.